lib/auth/init_test.go

```diff
@@ -737,6 +737,40 @@ func keysIn[K comparable, V any](m map[K]V) []K {
 	return result
 }
 
+type failingTrustInternal struct {
+	services.TrustInternal
+}
+
+func (t *failingTrustInternal) CreateCertAuthority(ctx context.Context, ca types.CertAuthority) error {
+	return trace.Errorf("error")
+}
+
+// TestInitCertFailureRecovery ensures the auth server is able to recover from
+// a failure in the cert creation process.
+func TestInitCertFailureRecovery(t *testing.T) {
```

Comment:
How is this test related to the fix? If we are using a fake `TrustInternal` implementation, aren't we faking something that _calls_ `CreateResources`? Does this test fail without the fix?

Reply:
The issue was multiple calls to `CreateResources` with the same set of resources would fail. This meant that if there was an error later in the init process that caused Teleport to crash and the init process the be retried it would get stuck on `CreateResources` on all subsequent attempts.

Its not a fake `TrustInternal`, I just wrap it to inject an error in cert creation to force a failure on the first init and then assert that a subsequent init is still able to succeed. And yes this test fails without this fix.